### PR TITLE
sys/uniqueid: Fixed rendundancy code and dependencies to multi support

### LINKED ISSUE #293: Needs make a coding convention with code redundancy
#### Description

It's important to create a coding convention about modules to avoid code redundancy when they need to call each other , for example:
-> uniqueid
-> radio
radio module has ```USEMODULE += uniqueid``` to import uniqueid.h and uniqueid module has ``` USEMODULE += radio``` to import radio.h and this cause redundancy.

### PR DESCRIPTION
**Pull request by eduazocar**
### Contribution description
This Pr add support to uniqueid in native boards, and use the `periph_hwrng` to provide a random `uint32_t` value. So the use of netopt `NETOPT_RANDOM`will give it to the boards with radio integrated.  The radio module will be loaded only when the board has an radio interface, if this interface doesn't exist, will use the `periph_hwrng` to provide an random value of uniqueid. Also removes the uniqueid dependency from the radio module, this itself doesn't need pre-load to use it.

### Testing procedure
in the section uniqueid you have to configurate to the random, for see the random number generator running.
```
make menuconfig -C tests/uniqueid flash term
```

### Issues/PRs references
Fixes #293

### CODE REVIEW

**Review by RocioRojas — CHANGES_REQUESTED** [review]

**Review by RocioRojas — CHANGES_REQUESTED** [review]

File: tests/uniqueid/whitelist [hunk 1]
```diff
@@ -0,0 +1,2 @@
+m4a-24g
```
[the remark this answers is not in the document]
Comment (eduazocar) [reply tests/uniqueid/whitelist:1]: that it's only a register, i could remove it

File: tests/uniqueid/main.c [hunk 2]
```diff
@@ -19,7 +19,9 @@
  *
  * @author  xkevin190 <kevinvelasco193@gmail.com>
  * @author  RocioRojas <rociorojas391@gmail.com>
+ * @author  eduazocar <eduazocarv@gmail.com>
```
[the remark this answers is not in the document]
Comment (eduazocar) [reply tests/uniqueid/main.c:22]: When I run native this throw an error

**Comment by RocioRojas** [comment]
Looks good for me, do squash!!

File: tests/uniqueid/main.c [hunk 3]
```diff
@@ -68,6 +71,8 @@ void test_get_ipv6Address(void) {
     printf("\nSecond random IPv6\n");
     ipv6_addr_print(&output2);
     TEST_ASSERT_EQUAL_INT(0, ipv6_addr_equal(&output1, &output2));
+    printf("Binary data printed\n");
```
Comment (RocioRojas) [inline tests/uniqueid/main.c:74]: Please put  ```\n \n Binary data printed\n```

**Review by RocioRojas — APPROVED** [review]

**Pull request merged**